### PR TITLE
Update to NetStandard 2.0

### PR DESCRIPTION
**Pull request by joebeernink**
Updated DotNetty.Common to .NETStandard 2.0 to allow this code to be integrated with projects targeting newer versions of the Microsoft.Extensions.* nugets and newer versions of Google.Protobuf

Eliminated Support for .NET45
Eliminated Support for .NETStandard1.3
Updated all nugets consumed to latest version

Ran All Unit Tests:
image

### CODE REVIEW

File: src/DotNetty.Codecs/DotNetty.Codecs.csproj [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-    <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
     <IsPackable>true</IsPackable>
     <PackageId>DotNetty.Codecs</PackageId>
     <Description>General purpose codecs for DotNetty</Description>
     <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
     <AssemblyTitle>DotNetty: codecs</AssemblyTitle>
     <NeutralLanguage>en-US</NeutralLanguage>
-    <VersionPrefix>0.6.0</VersionPrefix>
+    <VersionPrefix>2.0.0</VersionPrefix>
```
Comment (nayato) [inline src/DotNetty.Codecs/DotNetty.Codecs.csproj]: let's go for 0.7. For 0.x, incrementing "x" implies breaking changes (which these are). 2.0 implies it is a version with support commitment and I doubt we want to do that.

File: src/DotNetty.Handlers/Tls/TlsHandler.cs [hunk 2]
```diff
@@ -718,7 +711,6 @@ public void ExpandSource(int count)
                 }
                 this.sslOwnedBuffer = default(ArraySegment<byte>);
 
-#if NETSTANDARD1_3
```
Comment (nayato) [inline src/DotNetty.Handlers/Tls/TlsHandler.cs:721]: SslStream API necessary for efficient interaction in synchronous context are present in .net standard 2.0 (unlike in 1.3). So, for TlsHandler.cs specifically, we should use `#else` branch instead now.

File: src/DotNetty.Transport/Channels/Sockets/AbstractSocketByteChannel.cs [hunk 3]
```diff
@@ -153,22 +153,8 @@ public override void FinishRead(SocketChannelAsyncOperation operation)
         protected override void ScheduleSocketRead()
         {
             SocketChannelAsyncOperation operation = this.ReadOperation;
-            bool pending;
-#if NETSTANDARD1_3
-            pending = this.Socket.ReceiveAsync(operation);
-#else
-            if (ExecutionContext.IsFlowSuppressed())
```
Comment (nayato) [inline src/DotNetty.Transport/Channels/Sockets/AbstractSocketByteChannel.cs:160]: `ExecutionContext` was not available .net standard 1.3. It is in 2.0 so we should keep `#else` branch here (and elsewhere with `ExecutionContext`) instead.

File: test/DotNetty.Handlers.Tests/MediationStream.cs [hunk 4]
```diff
@@ -39,38 +39,6 @@ public override void SetLength(long value)
 
         public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) => this.writeDataFunc(new ArraySegment<byte>(buffer, offset, count));
 
-#if !NETCOREAPP1_1
```
Comment (nayato) [inline test/DotNetty.Handlers.Tests/MediationStream.cs]: pls revert per suggestion in TlsHandler.

File: examples/Discard.Client/Discard.Client.csproj [hunk 5]
```diff
@@ -1,13 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp2.0;net451</TargetFrameworks>
-    <NetStandardImplicitPackageVersion Condition=" '$(TargetFramework)' == 'netcoreapp2.0' ">1.6.1</NetStandardImplicitPackageVersion>
+    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
```
Comment (drwill-ms) [inline examples/Discard.Client/Discard.Client.csproj:4]: The .NET SDK still needs the older targets too. Can you make this additive only?

**Pull request closed without merging**